instructor/compiler.py

```diff
@@ -303,39 +303,34 @@ def facs_kinds(assignment):
 
 def micro_model(a):
     ret = {'is_ab': True}
-    for sp in a.microscopy_sample_prep.all():
-        for i in sp.microscopy_images.all():
-            key = "{sp.analysis}%%{sp.condition}%%SP_ID{protocol}".format(
-                sp=sp,
-                protocol=i.strain_protocol_id
+    for sample_prep in a.microscopy_sample_prep.all():
+        for mapping in sample_prep.image_mapping.all():
+            key = "{sp.micro_analysis}%%{sp.condition}%%SP_ID_{protocol}".format(
+                sp=sample_prep,
+                protocol=mapping.strain_protocol.id
             )
-            if key not in ret:
-                ret[key] = {
-                    'slides': [
+            if mapping.images.count() > 0:
+                if key not in ret:
+                    ret[key] = {
+                        'slides': [],
+                        'slide_type': sample_prep.micro_analysis
+                    }
+
+                for image in mapping.images.all():
+                    ret[key]['slides'].append(
                         {
-                            'hash': "IMAGE_{}".format(i.pk),
-                            'if_type': i.filter,
-                            'mag': i.objective
+                            'hash': image.pk,
```

Comment:
the hash should be just the pk of the image. `if_type` is for the filters, that are not implemented yet.